tests/functions/test_user_defined_func.py

```diff
@@ -109,80 +140,67 @@ def myFunction(variable):
         val = benchmark(e, [0])
         assert val == 1.0
 
-    @pytest.mark.parametrize("op", [ # parameter is string since compiled udf doesn't support closures as of present
-                        "Eq",
-                        "NotEq",
-                        "Lt",
-                        "LtE",
-                        "Gt",
-                        "GtE",
+    @pytest.mark.parametrize("op,var1,var2,expected", [ # parameter is string since compiled udf doesn't support closures as of present
+                        ("Eq", 1.0, 2.0, 0.0),
+                        ("Eq", [1.0, 2.0], [1.0, 2.0], [1.0, 1.0]),
+                        ("Eq", 1.0, [1.0, 2.0], [1.0, 0.0]),
+                        ("Eq", [2.0, 1.0], 1.0, [0.0, 1.0]),
+                        ("NotEq", 1.0, 2.0, 1.0),
+                        ("NotEq", [1.0, 2.0], [1.0, 2.0], [0.0, 0.0]),
+                        ("NotEq", 1.0, [1.0, 2.0], [0.0, 1.0]),
+                        ("NotEq", [2.0, 1.0], 1.0, [1.0, 0.0]),
+                        ("Lt", 1.0, 2.0, 1.0),
+                        ("Lt", [1.0, 2.0], [1.0, 2.0], [0.0, 0.0]),
+                        ("Lt", 1.0, [1.0, 2.0], [0.0, 1.0]),
+                        ("Lt", [2.0, 1.0], 1.0, [0.0, 0.0]),
+                        ("LtE", 1.0, 2.0, 1.0),
+                        ("LtE", [1.0, 2.0], [1.0, 2.0], [1.0, 1.0]),
+                        ("LtE", 1.0, [1.0, 2.0], [1.0, 1.0]),
+                        ("LtE", [2.0, 1.0], 1.0, [0.0, 1.0]),
+                        ("Gt", 1.0, 2.0, 0.0),
+                        ("Gt", [1.0, 2.0], [1.0, 2.0], [0.0, 0.0]),
+                        ("Gt", 1.0, [1.0, 2.0], [0.0, 0.0]),
+                        ("Gt", [2.0, 1.0], 1.0, [1.0, 0.0]),
+                        ("GtE", 1.0, 2.0, 0.0),
+                        ("GtE", [1.0, 2.0], [1.0, 2.0], [1.0, 1.0]),
+                        ("GtE", 1.0, [1.0, 2.0], [1.0, 0.0]),
+                        ("GtE", [2.0, 1.0], 1.0, [1.0, 1.0]),
                         ])
     @pytest.mark.parametrize("bin_execute", ['Python',
                                              pytest.param('LLVM', marks=pytest.mark.llvm),
                                              pytest.param('PTX', marks=[pytest.mark.llvm, pytest.mark.cuda]),
                                             ])
     @pytest.mark.benchmark(group="Function UDF")
-    def test_user_def_func_cmpop(self, op, bin_execute, benchmark):
+    def test_user_def_func_cmpop(self, op, var1, var2, expected, bin_execute, benchmark):
```

Comment:
This used to test Pyhton comparison operators, but was switched to numpy.
It also combines testing of comparison an `as_type` in one tests.
Replacing the lines with `1.0 if np.op(var1, var2) else 0.0` would be more directed.

the ops should also use space after comman consistently.

Reply:
> This used to test Pyhton comparison operators, but was switched to numpy.
Good catch - I'll split the tests so there is one for numpy and one for normal comparisons

Replacing with `1.0 if np.op(var1, var2) else 0.0` isn't possible, as numpy comparison ops return an array where each op is done elementwise (e.g. `np.equal([1,0], [1,2])` returns `[True, False]`). The astype call here is needed to convert the resulting boolean array into something that can be returned from compiled mode.

Reply:
makes sense, ifexp is better left for later